keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterSpec.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ *
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.keel.api.ec2
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.netflix.spinnaker.keel.api.ec2.image.ImageProvider
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+data class ClusterSpec(
```

Comment:
Let me explore that. I agree this way is icky.

Reply:
My only concern is we might end up with a ton of generic types on `Cluster`. Worth exploring though.

Reply:
(as discussed offline)

I explored using generics/inheritance but it seemed like it might quickly become really gross. I extracted the common data classes to their own files so the typing looked better instead.

I'm thinking that we may soon have more linked/overriden properties. Let's explore how to make this better at that point (unless @cfieber or @asher have any great ideas??)

Reply:
I don't have a better suggestion and agree lets figure it out when we have more linkages